runtime/memory/cortex_m.c

```diff
@@ -32,6 +32,19 @@ void expand_memory() {
     memory_size += WASM_PAGE_SIZE;
 }
 
+i32 instruction_memory_size() {
+    return memory_size / WASM_PAGE_SIZE;
+}
+
+i32 instruction_memory_grow(i32 count) {
+    i32 prev_size = instruction_memory_size();
+    for (int i = 0; i < count; i++) {
+        expand_memory();
+    }
+
+    return prev_size;
+}
```

Comment:
This seems very unlikely to work on cortex_m

Reply:
The cortex_m implementation of expand_memory does move the memory limit it seems like. If it runs out of space in the configured buffer it will assert.